compiler/src/dotty/tools/dotc/typer/ProtoTypes.scala

```diff
@@ -48,7 +48,7 @@ object ProtoTypes {
 
     private def disregardProto(pt: Type)(implicit ctx: Context): Boolean = pt.dealias match {
       case _: OrType => true
-      case pt => pt.isRef(defn.UnitClass)
+      case ptd => ptd.isRef(defn.UnitClass)
```

Comment:
I think the point here is to specifically shadow it, we do this all over the compiler :)